geopandas/geoseries.py

```diff
@@ -291,4 +350,13 @@ def __sub__(self, other):
         """Implement - operator as for builtin set type"""
         return self.difference(other)
 
+    def _reindex_indexer(self, new_index, indexer, copy):
+        if indexer is None:
```

Comment:
Can you put a comment in the docstring this is overwriting one in Series (and can also add why it is needed)